auto_round/schemes.py

```diff
@@ -0,0 +1,176 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Optional
+
+__all__ = ["QuantizationScheme", "preset_name_to_scheme"]
+
+
+@dataclass
+class QuantizationScheme:
+    bits: int = 4
+    group_size: int = 128
+    sym: bool = True
+    data_type: str = "int"
+    act_bits: Optional[int] = None
+    act_group_size: Optional[int] = None
+    act_sym: Optional[bool] = None
+    act_data_type: Optional[str] = None
+    act_dynamic: Optional[bool] = None
+    super_bits: Optional[int] = None
+    super_group_size: Optional[int] = None
+
+    @classmethod
+    def from_dict(cls, config: dict):
+        return cls(**config)
+
+
+def preset_name_to_scheme(name: str) -> QuantizationScheme:
+    """Get a QuantizationScheme instance from a preset scheme name."""
+    name = name.upper()
+
+    if name not in PRESET_SCHEMES:
+        raise KeyError(f"Unknown preset scheme name {name}, " f"available names: {list(PRESET_SCHEMES.keys())}")
+
+    scheme_args = deepcopy(PRESET_SCHEMES[name])
+    return scheme_args
+
+
+def is_preset_scheme(name: str) -> bool:
+    """Check if the given name is a preset scheme name."""
+    return name.upper() in PRESET_SCHEMES
+
+
+W4A16 = QuantizationScheme.from_dict(
+    {
+        "bits": 4,
+        "sym": True,
+        "group_size": 128,
+        "data_type": "int",
+        "act_bits": 16,
+    }
+)
+
+W2A16 = QuantizationScheme.from_dict(
+    {
+        "bits": 2,
+        "sym": True,
+        "group_size": 128,
+        "data_type": "int",
+        "act_bits": 16,
+    }
+)
+
+W3A16 = QuantizationScheme.from_dict(
+    {
+        "bits": 3,
+        "sym": True,
+        "group_size": 128,
+        "data_type": "int",
+        "act_bits": 16,
+    }
+)
+
+W8A16 = QuantizationScheme.from_dict(
+    {
+        "bits": 8,
+        "sym": True,
+        "group_size": 128,
+        "data_type": "int",
+        "act_bits": 16,
+    }
+)
+
+MXFP4 = QuantizationScheme.from_dict(
+    {
+        "bits": 4,
+        "group_size": 32,
+        "data_type": "mx_fp",
+        "act_bits": 4,
+        "act_data_type": "mx_fp_rceil",
+    }
+)
+
+MXFP8 = QuantizationScheme.from_dict(
+    {
+        "bits": 8,
+        "group_size": 32,
+        "data_type": "mx_fp",
+        "act_bits": 8,
+        "act_data_type": "mx_fp_rceil",
+    }
+)
+
+NVFP4 = QuantizationScheme.from_dict(
+    {
+        "bits": 4,
+        "group_size": 16,
+        "data_type": "nv_fp",
+        "act_bits": 4,
+        "act_data_type": "nv_fp_with_static_gs",
```

Comment:
similar as above. static_gs does not belong to data type.

how about adding an act_scheme field? it can be static_gs, rceil, etc

Reply:
Theoretically, you are correct. However, each dtype has its own scheme, for example, FP8 may use  unit scale. As the number of supported dtypes grows, the number of keys like act_scheme also increases. Therefore, combining certain schemes with the data_type is a more practical approach for coding.